### PR TITLE
Fix update field_attributes for Account Model

### PR DESCRIPTION
**Pull request by hongthang152**
#9611 
Remove checking for Hash object, change to checking for Array object.

### CODE REVIEW

File: app/models/account.rb [hunk 1]
```diff
@@ -230,8 +230,8 @@ def fields_attributes=(attributes)
     fields     = []
     old_fields = self[:fields] || []
 
-    if attributes.is_a?(Hash)
-      attributes.each_value do |attr|
+    if attributes.is_a?(Array)
```
Comment (Gargron) [inline app/models/account.rb]: This change isn't right either, because we get a hash from a standard form submission. So it can be a hash or an array.

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/models/account.rb (hunk 1 above)
Comment (hongthang152) [reply app/models/account.rb]: Change to 
`Array.wrap(attributes).each do |attr|`

We wrap the array around a single element to allow iteration.

Comment (hongthang152) [reply app/models/account.rb]: @Gargron You can check the latest commit

File: app/models/account.rb [hunk 2]
```diff
@@ -230,18 +230,16 @@ def fields_attributes=(attributes)
     fields     = []
     old_fields = self[:fields] || []
 
-    if attributes.is_a?(Hash)
-      attributes.each_value do |attr|
-        next if attr[:name].blank?
+    Array.wrap(attributes).each do |attr|
+      next if !attr.is_a?(Hash) || attr[:name].blank?
```
Comment (nightpool) [inline app/models/account.rb]: we should throw a validation error here instead of passing silently, to resolve #9611

Comment (hongthang152) [reply app/models/account.rb]: It raises error now when it fails the validation.

Comment (hongthang152) [reply app/models/account.rb]:
```
     if !attr.is_a?(Hash) || attr[:name].blank?
        raise ArgumentError, 'One of custom field attributes has invalid format'
      end
```

File: app/models/account.rb [hunk 3]
```diff
@@ -230,18 +230,17 @@ def fields_attributes=(attributes)
...
-        next if attr[:name].blank?
-
-        previous = old_fields.find { |item| item['value'] == attr[:value] }
-
-        if previous && previous['verified_at'].present?
-          attr[:verified_at] = previous['verified_at']
-        end
+    Array.wrap(attributes).each do |attr|
+      if !attr.is_a?(Hash) || attr[:name].blank?
+        raise ArgumentError, 'One of custom field attributes has invalid format'
```
Comment (Gargron) [inline app/models/account.rb]: I'm sorry, I disagree with @nightpool on this... Raising ArgumentError will lead to error 500 in the API, so it won't be very useful. It's possible to raise ActiveRecord::RecordInvalid, but I am not 100% sure if this is the right place to do it (this is in a setter, not in the validation cycle...)

Until there is a better validation solution, silently ignoring seems better. Sorry for long delay.

**Review by Gargron — CHANGES_REQUESTED** [review]

**Pull request closed without merging**